core/state_transition.go

```diff
@@ -438,167 +426,98 @@ func (st *StateTransition) applyEditValidatorTx(
 		wrapper.Validator.UpdateHeight = blockNum
 	}
 
-	if newRate.Sub(rateAtBeginningOfEpoch).Abs().GT(wrapper.Validator.MaxChangeRate) {
-		return errCommissionRateChangeTooFast
-	}
-
-	if newRate.GT(wrapper.Validator.MaxRate) {
-		return errCommissionRateChangeTooHigh
-	}
-
-	if err := st.state.UpdateStakingInfo(wrapper.Address, wrapper); err != nil {
-		return err
-	}
-	return nil
+	return st.state.UpdateStakingInfo(wrapper.Address, wrapper)
 }
 
-func (st *StateTransition) applyDelegateTx(delegate *staking.Delegate) error {
-	if delegate.Amount.Sign() == -1 {
-		return errNegativeAmount
-	}
-
-	if !st.state.IsValidator(delegate.ValidatorAddress) {
-		return errValidatorNotExist
+func (st *StateTransition) verifyAndApplyDelegateTx(delegate *staking.Delegate) error {
+	err := VerifyDelegateStakingMsg(st.state, delegate)
```

Comment:
similar. We basically don't want to duplicate the complicated validator object manipulation logic twice, it's error-prone. Just do it once and use the modified object here to commit.